tailwind.config.js

```diff
@@ -1,4 +1,7 @@
 /* eslint-disable no-undef, @typescript-eslint/no-var-requires */
 const config = require("./jslib/components/tailwind.config.base");
 
-module.exports = config;
+module.exports = {
+  ...config,
+  darkMode: "class",
+};
```

Comment:
We unfortunately can't use tailwind dark since we need to support multiple themes on some clients. @danielleflinn and my approach has been to use css variables for colors.

Reply:
For svg icons we used `grayscale` modifier, and `currentColor` to tweak things.

Reply:
One possibility is to make an svg template per https://angular.io/guide/svg-in-templates

Reply:
I went with the SVG template approach. The colours do not seem to align particularly with our theming, so I hard-coded them into the component rather than trying to export them from the css. There's also a bit of logic to handle theme changes, ideally this would be abstracted but the `platformUtilsService.onDefaultSystemThemeChange` method doesn't provide any way to deregister the event handler. I should create a `ThemeService` to pull this out of `PlatformUtilsService` and provide this functionality, but I haven't done that at this stage.

Reply:
https://user-images.githubusercontent.com/31796059/165219798-a30c03b1-0348-4898-a5fb-cd8bd36b5219.mov

Reply:
For context to my comment below. We would like to avoid hard-coding color variants since it makes looking them up fragmented, and will making porting this component to other clients a chore.

Reply:
Discussion continued in your comment below